src/runner.sh

```diff
@@ -123,6 +123,12 @@ function runner::run_test() {
   if [[ $test_result_code -ne 0 ]]; then
     state::add_tests_failed
     console_results::print_error_test "$function_name" "$test_result_code"
+
+    if [ "$_STOP_ON_FAILURE" = true ]; then
+      console_results::print_failed_test "$function_name" "$test_result_code"
+      exit 1
```

Comment:
Would it make sense to disable the exit trap before the exit 1, @Tito-Kati?
```suggestion
      trap '' EXIT && exit 1
```

Reply:
Ok, we can do this change later if we see the need. We don't need to do it in this PR.